src/deployment.js

```diff
@@ -35,48 +39,33 @@ class Deployment {
     try {
       core.info(`Actor: ${this.buildActor}`)
       core.info(`Action ID: ${this.actionsId}`)
-      const pagesDeployEndpoint = `${this.githubApiUrl}/repos/${this.repositoryNwo}/pages/deployment`
-      const artifactExgUrl = `${this.runTimeUrl}_apis/pipelines/workflows/${this.workflowRun}/artifacts?api-version=6.0-preview`
-      core.info(`Artifact URL: ${artifactExgUrl}`)
-      const { data } = await axios.get(artifactExgUrl, {
-        headers: {
-          Authorization: `Bearer ${this.runTimeToken}`,
-          'Content-Type': 'application/json'
-        }
+      core.info(`Actions Workflow Run ID: ${this.workflowRun}`)
+
+      const artifactUrl = await getSignedArtifactUrl({
+        runtimeToken: this.runTimeToken,
+        workflowRunId: this.workflowRun,
+        artifactName: this.artifactName
       })
-      core.info(JSON.stringify(data))
-      const artifactRawUrl = data?.value?.find(artifact => artifact.name === this.artifactName)?.url
-      if (!artifactRawUrl) {
-        throw new Error(
-          'No uploaded artifact was found! Please check if there are any errors at build step, or uploaded artifact name is correct.'
-        )
-      }
 
-      const artifactUrl = `${artifactRawUrl}&%24expand=SignedContent`
-      const payload = {
-        artifact_url: artifactUrl,
-        pages_build_version: this.buildVersion,
-        oidc_token: idToken
-      }
-      if (this.isPreview === true) {
-        payload.preview = true
-      }
-      core.info(`Creating deployment with payload:\n${JSON.stringify(payload, null, '\t')}`)
-      const response = await axios.post(pagesDeployEndpoint, payload, {
-        headers: {
-          Accept: 'application/vnd.github.v3+json',
-          Authorization: `Bearer ${this.githubToken}`,
-          'Content-type': 'application/json'
-        }
+      const deployment = await createPagesDeployment({
+        githubToken: this.githubToken,
+        artifactUrl,
+        buildVersion: this.buildVersion,
+        idToken,
+        isPreview: this.isPreview
       })
-      this.requestedDeployment = true
-      core.info(`Created deployment for ${this.buildVersion}`)
-      if (response && response.data) {
-        core.info(JSON.stringify(response.data))
-        this.deploymentInfo = response.data
+      this.deploymentInfo = {
+        ...deployment,
+        id: deployment?.['status_url']?.split('/')?.pop() || this.buildVersion,
+        pending: true
```

Comment:
Does it matter that in the previous version we only set `this.deploymentInfo` when `response.data` is present, and in this version we set it all the time with possibly undefined values?

Reply:
With Octokit, we should always have `response.data` (as `deployment`, in this case) when the request returns successfully. Error (`>= 400` status code) responses should result in the Promise being rejected and hitting the `catch` logic branch.

https://github.com/octokit/request.js/blob/b432f47bdd456d3806e24378caba6ac889cb7d72/src/fetch-wrapper.ts#L102-L116

AFAIK, that was also true of Axios, so I'm not 100% on the exact choices (probably misunderstanding or caution 🤷🏻) that went into the previously used logic here.

Reply:
I'm going with caution, mainly because we're not in TS land and `response.data` could be undefined if some internal method doesn't create `response` properly. Not saying we need to change this, but I think that's why it was written that way.

Reply:
Eh, putting it back in doesn't really add any cognitive overhead, so playing it safe makes sense. 👍🏻

Reply:
Added an `if` condition back in for this. 👍🏻